gnmi_server/gnoi.go

```diff
@@ -20,6 +20,40 @@ import (
 	jwt "github.com/dgrijalva/jwt-go"
 )
 
+func KillOrRestartProcess(restart bool, serviceName string) error {
+	sc, err := ssc.NewDbusClient()
+	if err != nil {
+		return err
+	}
+	if restart {
+		log.V(2).Infof("Restarting service %s...", serviceName)
+		err = sc.RestartService(serviceName)
+		if err != nil {
+			log.V(2).Infof("Failed to restart service %s: %v", serviceName, err)
+		}
+	} else {
+		log.V(2).Infof("Stopping service %s...", serviceName)
+		err = sc.StopService(serviceName)
+		if err != nil {
+			log.V(2).Infof("Failed to stop service %s: %v", serviceName, err)
+		}
+	}
+	return err
+}
+
+func (srv *Server) KillProcess(ctx context.Context, req *gnoi_system_pb.KillProcessRequest) (*gnoi_system_pb.KillProcessResponse, error) {
+	serviceName := req.GetName()
```

Comment:
>serviceName [](http://example.com/codeflow?start=1&length=11)

Where is the code to check serviceName is allowed? For example, kill or restart gnmi or syncd is problematic.

Reply:
Checking if serviceName is allowed is done in dbus: https://github.com/sonic-net/sonic-host-services/blob/fa6fc0f2972a8ad10ca3a2dfcb8e9e93f5c24ad8/host_modules/systemd_service.py#L20

This error message is passed to gnmi_client. This scenario is tested in sonic-mgmt through this PR: https://github.com/sonic-net/sonic-mgmt/pull/12478/files